### ISSUE TITLE
Nginx VRL parsing function does not use default Nginx log format

### ISSUE THREAD

**Issue by awcrotwell** [timberio/vector#7236]
### Vector Version

```
vector 0.13.0 (v0.13.0 x86_64-unknown-linux-gnu 2021-04-21)
```

### Vector Configuration File

```toml
[sources.test_source]
  type = "file"
  include = ["/home/andrew/workspace/js/dendro-cli/src/vector/.logs/test.log"]
  read_from = "beginning"
  ignore_checkpoints = true

[transforms.test_transform]
  type = "remap"
  inputs = ["test_source"]
  source = '''
  .log = parse_nginx_log!(.message, "combined")
  '''

[sinks.test_sink]
  # General
  type = "console" # required
  inputs = ["test_transform"] # required
  target = "stdout" # optional, default

  # Encoding
  encoding.codec = "json" # required
```

### Debug Output

```
Apr 23 13:21:42.243  INFO vector::app: Log level is enabled. level="vector=trace,codec=trace,vrl=trace,file_source=trace,tower_limit=trace,rdkafka=trace"
Apr 23 13:21:42.243  INFO vector::sources::host_metrics: PROCFS_ROOT is unset. Using default '/proc' for procfs root.
Apr 23 13:21:42.243  INFO vector::sources::host_metrics: SYSFS_ROOT is unset. Using default '/sys' for sysfs root.
Apr 23 13:21:42.243  INFO vector::app: Loading configs. path=[("vector-config.toml", None)]
Apr 23 13:21:42.247  INFO vector::topology: Running healthchecks.
Apr 23 13:21:42.247  INFO vector::topology: Starting source. name="test_source"
Apr 23 13:21:42.247  INFO vector::topology::builder: Healthcheck: Passed.
Apr 23 13:21:42.247  INFO vector::topology: Starting transform. name="test_transform"
Apr 23 13:21:42.247  INFO vector::topology: Starting sink. name="test_sink"
Apr 23 13:21:42.247  INFO source{component_kind="source" component_name=test_source component_type=file}: vector::sources::file: Starting file server. include=["/home/andrew/workspace/js/dendro-cli/src/vector/.logs/test.log"] exclude=[]
Apr 23 13:21:42.247 DEBUG transform{component_kind="transform" component_name=test_transform component_type=remap}: vector::utilization: utilization=0.9714908796811138
Apr 23 13:21:42.247  INFO vector: Vector has started. version="0.13.0" git_version="v0.13.0" released="Wed, 21 Apr 2021 17:18:41 +0000" arch="x86_64"
Apr 23 13:21:42.247  INFO vector::app: API is disabled, enable by setting `api.enabled` to `true` and use commands like `vector top`.
Apr 23 13:21:42.248  INFO source{component_kind="source" component_name=test_source component_type=file}:file_server: file_source::checkpointer: Loaded checkpoint data.
Apr 23 13:21:42.248  INFO source{component_kind="source" component_name=test_source component_type=file}:file_server: vector::internal_events::file::source: Found new file to watch. path=/home/andrew/workspace/js/dendro-cli/src/vector/.logs/test.log
Apr 23 13:21:42.248 TRACE vector: Beep.
Apr 23 13:21:42.249 TRACE source{component_kind="source" component_name=test_source component_type=file}:file_server: file_source::file_server: Continue watching file. path="/home/andrew/workspace/js/dendro-cli/src/vector/.logs/test.log"
Apr 23 13:21:42.248 DEBUG sink{component_kind="sink" component_name=test_sink component_type=console}: vector::utilization: utilization=0.038989531481957274
Apr 23 13:21:42.249 TRACE source{component_kind="source" component_name=test_source component_type=file}:file_server: file_source::file_server: Read bytes. path="/home/andrew/workspace/js/dendro-cli/src/vector/.logs/test.log" bytes=238
Apr 23 13:21:42.249 TRACE source{component_kind="source" component_name=test_source component_type=file}: vector::internal_events::file::source: Received one event. file=/home/andrew/workspace/js/dendro-cli/src/vector/.logs/test.log byte_size=238
Apr 23 13:21:42.250  WARN transform{component_kind="transform" component_name=test_transform component_type=remap}: vector::internal_events::remap: Mapping failed with event. error="function call error for \"parse_nginx_log\" at (9:47): failed parsing log line" internal_log_rate_secs=30
{"file":"/home/andrew/workspace/js/dendro-cli/src/vector/.logs/test.log","host":"andrew-xps159560","message":"0.0.0.0 - - [23/Apr/2021:14:59:24 +0000] \"GET /my-path/manifest.json HTTP/1.1\" 200 504 \"https://my-url.com/my-path\" \"Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/89.0.4389.90 Safari/537.36\"","source_type":"file","timestamp":"2021-04-23T17:21:42.249155496Z"}
Apr 23 13:21:43.249 TRACE vector: Beep.
Apr 23 13:21:43.255 DEBUG vector::internal_events::file::source: Files checkpointed. count=3 duration_ms=5
Apr 23 13:21:43.281 DEBUG source{component_kind="source" component_name=test_source component_type=file}:file_server: file_source::file_server: event_throughput=1.000/sec bytes_throughput=238.000/sec ratios={"discovery": 0.00037978307, "other": 0.00010705757, "reading": 0.000059049144, "sending": 0.0000477682, "sleeping": 0.99940693}
Apr 23 13:21:43.282 TRACE source{component_kind="source" component_name=test_source component_type=file}:file_server: file_source::file_server: Continue watching file. path="/home/andrew/workspace/js/dendro-cli/src/vector/.logs/test.log"
Apr 23 13:21:43.735  INFO vector: Vector has stopped.
```

### Expected Behavior

My `test.log` is holding raw nginx logs. The expected behavior is the `parse_nginx_log!` will work.

### Actual Behavior

The `parse_nginx_log!` function call does nothing.

### Example Data

My data: 
```
0.0.0.0 - - [23/Apr/2021:14:59:24 +0000] "GET /my-path/manifest.json HTTP/1.1" 200 504 "https://my-url.com/my-path" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/89.0.4389.90 Safari/537.36"
```

Example data from https://vector.dev/docs/reference/vrl/functions/#examples-49
```
172.17.0.1 alice - [01/Apr/2021:12:02:31 +0000] "POST /not-found HTTP/1.1" 404 153 "http://localhost/somewhere" "Mozilla/5.0 (Windows NT 6.1) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/72.0.3626.119 Safari/537.36" "2.75"
```

### Additional Context

I believe the vector `parse_nginx_log!` expects the final `"2.75"` provided in the example data. This is maps to the compression key. #6103 shows the default log and it does not contain the compression info. I believe the compression format comes from the second code block [found here](https://docs.nginx.com/nginx/admin-guide/monitoring/logging/).

```
$ vector vrl
```

Example data from https://vector.dev/docs/reference/vrl/functions/#examples-49
```
$ parse_nginx_log!(s'172.17.0.1 alice - [01/Apr/2021:12:02:31 +0000] "POST /not-found HTTP/1.1" 404 153 "http://localhost/somewhere" "Mozilla/5.0 (Windows NT 6.1) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/72.0.3626.119 Safari/537.36" "2.75"', "combined")
{ "agent": "Mozilla/5.0 (Windows NT 6.1) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/72.0.3626.119 Safari/537.36", "client": "172.17.0.1", "compression": "2.75", "method": "POST", "path": "/not-found", "protocol": "HTTP/1.1", "referer": "http://localhost/somewhere", "request": "POST /not-found HTTP/1.1", "size": 153, "status": 404, "timestamp": t'2021-04-01T12:02:31Z', "user": "alice" }
```

My data:
```
$ parse_nginx_log!(s'0.0.0.0 - - [23/Apr/2021:14:59:24 +0000] "GET /my-path/manifest.json HTTP/1.1" 200 504 "https://my-url.com/my-path" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/89.0.4389.90 Safari/537.36"', "combined")
function call error for "parse_nginx_log" at (0:271): failed parsing log line
```

### References

#6103 
#6973

**Comment by nnz1024** [comment]
Looks like problem is in "compression" field, which is not a part of default Nginx log format.
Documentation says that
```
log_format combined '$remote_addr - $remote_user [$time_local] '
                    '"$request" $status $body_bytes_sent '
                    '"$http_referer" "$http_user_agent"';
```

Please also note that mandatory dash must be between `$remote_addr` and `$remote_user`, not after `$remote_user`, as #6973 suggests. I even checked it on Nginx 1.14.0 and got
```
aa.bb.cc.dd - testuser [24/Apr/2021:02:13:26 +0300] "GET / HTTP/1.1" 200 396 "-" "Mozilla/5.0 (X11; Linux x86_64; rv:78.0) Gecko/20100101 Firefox/78.0"
```

Nothing about compression, and username is the third parameter, not second.

**Comment by nnz1024** [comment]
BTW, Nginx error log parsing is also non-ideal: it cannot parse messages without connection ID
```
$ parse_nginx_log!(s'2021/04/24 02:12:58 [notice] 19319#19319: signal process started', "error")
function call error for "parse_nginx_log" at (0:94): failed parsing log line

$ parse_nginx_log!(s'2021/04/24 02:13:26 [error] 19381#19381: send() failed (111: Connection refused)', "error")
function call error for "parse_nginx_log" at (0:110): failed parsing log line
```

**Comment by JeanMertz** [comment]
fixed in #7852.
